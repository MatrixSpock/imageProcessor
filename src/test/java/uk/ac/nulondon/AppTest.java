package uk.ac.nulondon;

import org.approvaltests.Approvals;
import org.approvaltests.awt.AwtApprovals;
import org.assertj.core.api.Assertions;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import javax.imageio.ImageIO;
import java.awt.Color;
import java.awt.image.BufferedImage;
import java.io.File;
import java.io.IOException;
import java.util.List;

public class AppTest {

    Image image;

    @BeforeEach
    void setup() throws IOException {
        BufferedImage bufferedImage = ImageIO.read(new File("src/main/resources/beach.png"));
        image = new Image(bufferedImage);
    }

    @Test
    void getGreenest() {
        Assertions.assertThat(image.getGreenest()).isEqualTo(7);
    }

    @Test
    void toBufferedImage() {
        AwtApprovals.verify(image.toBufferedImage());
    }

    @Test
    void highlightColumn() {
        List<Color> column = image.highlightColumn(3);
        Approvals.verify(column.toString());
        AwtApprovals.verify(image.toBufferedImage());
    }

    @Test
    void removeColumn() {
        List<Color> column = image.removeColumn(3);
        Approvals.verify(column.toString());
        AwtApprovals.verify(image.toBufferedImage());
    }

    @Test
    void insertColumn() {
        List<Color> column = List.of(
                Color.RED, Color.ORANGE, Color.YELLOW, Color.GREEN,
                Color.CYAN, Color.BLUE, Color.MAGENTA, Color.PINK);
        image.addColumn(2, column);
        AwtApprovals.verify(image.toBufferedImage());
    }
}
