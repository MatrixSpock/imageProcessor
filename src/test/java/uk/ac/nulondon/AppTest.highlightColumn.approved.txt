[java.awt.Color[r=153,g=217,b=234], java.awt.Color[r=153,g=217,b=234], java.awt.Color[r=168,g=230,b=29], java.awt.Color[r=168,g=230,b=29], java.awt.Color[r=153,g=217,b=234], java.awt.Color[r=77,g=109,b=243], java.awt.Color[r=77,g=109,b=243], java.awt.Color[r=245,g=228,b=156]]